lib/Dialect/OM/Evaluator/Evaluator.cpp

```diff
@@ -83,6 +83,9 @@ circt::om::Evaluator::instantiate(
     if (!actualParamType)
       return cls.emitError("actual parameter for ")
              << formalParamName << " is null";
+    if (isa<IntegerType>(actualParamType) &&
+        isa<om::OMIntegerType>(formalParamType))
+      continue;
```

Comment:
I feel it would be better to keep the strict equivalence here. For example in evaluation of Map, we look up key from `DenseMap<Attribute, Value>`. And therefore type is a part of the key information so I think it's very tricky to lose type equivalence in instantiation. I think actualParamType should have instead om::OMIntegerType and we should wrap integer in python side. WDYT?